src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp

```diff
@@ -310,6 +312,62 @@ TEST(OVClassBasicTest, smoke_SetConfigAutoNoThrows) {
     OV_ASSERT_NO_THROW(ie.set_property(CommonTestUtils::DEVICE_AUTO, ov::hint::model_priority(ov::hint::Priority::HIGH)));
     OV_ASSERT_NO_THROW(value = ie.get_property(CommonTestUtils::DEVICE_AUTO, ov::hint::model_priority));
     EXPECT_EQ(value, ov::hint::Priority::HIGH);
+
+    // profiling test
+    bool flag;
+    const bool enableFlag = true;
+    OV_ASSERT_NO_THROW(ie.set_property(CommonTestUtils::DEVICE_AUTO, ov::enable_profiling(enableFlag)));
+    OV_ASSERT_NO_THROW(flag = ie.get_property(CommonTestUtils::DEVICE_AUTO, ov::enable_profiling));
+    ASSERT_EQ(enableFlag, flag);
+
+    // batching test
+    bool batching;
+    OV_ASSERT_NO_THROW(ie.set_property(CommonTestUtils::DEVICE_AUTO, ov::allow_auto_batching(enableFlag)));
+    OV_ASSERT_NO_THROW(batching = ie.get_property(CommonTestUtils::DEVICE_AUTO, ov::allow_auto_batching));
+    ASSERT_EQ(enableFlag, batching);
+
+    // perf hint test
+    ov::hint::PerformanceMode mode;
+    uint32_t num_requests;
+    OV_ASSERT_NO_THROW(ie.set_property(CommonTestUtils::DEVICE_AUTO, ov::hint::performance_mode(ov::hint::PerformanceMode::THROUGHPUT)));
+    OV_ASSERT_NO_THROW(mode = ie.get_property(CommonTestUtils::DEVICE_AUTO, ov::hint::performance_mode));
+    ASSERT_EQ(ov::hint::PerformanceMode::THROUGHPUT, mode);
+    ASSERT_THROW(ie.set_property(CommonTestUtils::DEVICE_AUTO, {{ov::hint::performance_mode.name(), "ABC"}}), ov::Exception);
+    ASSERT_NO_THROW(ie.set_property(CommonTestUtils::DEVICE_AUTO, ov::hint::num_requests(8)));
+    ASSERT_NO_THROW(num_requests = ie.get_property(CommonTestUtils::DEVICE_AUTO, ov::hint::num_requests));
+    ASSERT_EQ(8, num_requests);
+
+    std::string deviceName = CommonTestUtils::DEVICE_CPU;
```

Comment:
this is a file with common tests which is used for other plugins tests. We can have configuration when CPU is not available, so we should not hardcode CPU here